### DISCUSSION TITLE
Inheritance and document initialization interaction should be clarified in the documentation

### DISCUSSION THREAD

**Discussion by sshghoult** [BeanieODM/beanie#484] [Feature Request]
I believe one should add a warning-like paragraph about the behaviour of the Inherited, but not Initialized in the run-time Document, the parent of which had been initialized.

Specifically, the case I encountered and got stuck with goes as such: 
1. Define a base document `A`, **do not** specify `Settings.is_root = True`
2. Define a document `B(A)` with additional fields
3. At the start up of your application initialize the document `A`, but skip the `B`
4. Write the application code using the model `B`

The resulting application will properly interact with the document B and it's additional fields, saving and restoring the documents correctly. However, the document B database-wise is treated as the document A, which we initialized and assigned the `cls._document_settings.motor_collection` to. The class B retrieves the attributes from A as a common unitialized python class via mro when those are required, which leads to the described behaviour.
The BSON objects of B in the database do not contain the  _class_id field, as the `Settings._is_root` hasn't been set to `True`, and are placed in the same collection named after `A`. Therefore there is no difference between the objects of `A` and objects of `B` in the database, and therefore no difference overall.

Even though such behaviour may be desirable for some use-cases, the fact that it is **implicit**, rather than explicit, and not in any way mentioned in the community, seems to be a problem. Described solution works out of the box and at first glance seems to be working correctly and raises no questions. It's a bit later into the development (possibly after several production releases, during the work on migrations, in my case - where one has to initialize the models separately) that the pecularities of the initialization mechanic get discovered after a deep dive into the internals of the library.

Then one has to carefully migrate all of the objects to a new model with `Settings.is_root` turned on and both `A` and `B` initialized. It's great if `A` only served as a base class and all of the objects semantically belonged to the class `B`, but should `A` be a separate meaningful entity, and the information will be lost.

You might find yourself classifying the problem to be of the PEBMAC kind, but I believe we should write out such behaviour specifics. A newcomer has no knowledge on the details of the initialization procedure and the process of assigning the collection name to the document class, therefore more or less blindly trying out the approaches suggested in the docs. Subsequently, there is a certain possibility that they will follow my way intuitively (or motivated by the phrase "Beanie Documents support inheritance as any other Python classes.").

**Comment by roman-right** [comment]
Looks reasonable. Thank you
